application/common/models/User.php

```diff
@@ -313,20 +294,19 @@ public function fields(): array
     public function save($runValidation = true, $attributeNames = null)
     {
         if ($this->scenario === self::SCENARIO_UPDATE_PASSWORD) {
-            return $this->savePassword();
+            return $this->updatePassword();
         }
 
         return parent::save($runValidation, $attributeNames);
     }
 
-    private function savePassword(): bool
+    private function updatePassword(): bool
     {
         $transaction = ActiveRecord::getDb()->beginTransaction();
 
         try {
-            $this->password_hash = password_hash($this->password, PASSWORD_DEFAULT);
 
-            if (! $this->saveHistory()) {
+            if (! $this->savePassword()) {
                 return false;
```

Comment:
you don't, not sure about the internal handling by the dbms for these but nothing gets written until the `commit()` runs.